contracts/interfaces/IPoolManager.sol

```diff
@@ -162,9 +162,10 @@ interface IPoolManager is IERC1155 {
     function getCurrencyDelta(uint256 id, Currency currency) external view returns (int256);
 
     /// @notice All operations go through this function
+    /// @param lockTarget The address to call the callback on
     /// @param data Any data to pass to the callback, via `ILockCallback(msg.sender).lockCallback(data)`
     /// @return The data returned by the call to `ILockCallback(msg.sender).lockCallback(data)`
-    function lock(bytes calldata data) external returns (bytes memory);
+    function lock(address lockTarget, bytes calldata data) external returns (bytes memory);
```

Comment:
I thought your original idea was to add a new lock function, instead of modifying existing one. Why change of mind?

Reply:
i always planned on just modifying the existing one, not a huge benefit of having a whole separate lock entrypoint that I can see?

Reply:
Fair enough. Maybe tiny gas savings in calling `lock` with 1 param instead of 2 is the only benefit